File: Arrays & Strings/13. Roman to Integer.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> map;
        map['I'] = 1; map['V'] = 5; map['X'] = 10; map['L'] = 50;
        map['C'] = 100; map['D'] = 500; map['M'] = 1000;

        int res = 0;
        for (int i = 0; i < s.size(); i++) {
            if (map[s[i]] < map[s[i + 1]]) {
                res -= map[s[i]];
            } else {
                res += map[s[i]];
            }
        }

        return res;
    }
};


int main() {
    string s = "IV";
    Solution sol;
    cout << sol.romanToInt(s) << endl;
}

File: Arrays & Strings/13. Roman to Integer.py
class Solution:
    def romanToInt(self, s: int) -> int:
        hmap = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        res = 0
        for i in range(len(s)):
            if (i < (len(s) - 1) and hmap[s[i]] < hmap[s[i+1]]):
               res -= hmap[s[i]]
            else:
                res += hmap[s[i]]
        return res


if __name__ == "__main__":
    s = input()
    sol = Solution()
    print(sol.romanToInt(s))
